pytoda/preprocessing/crawlers.py

```diff
@@ -127,3 +131,58 @@ def get_smiles_from_pubchem(
                 logger.warninig(f'Did not find any result for drug: {drug}')
                 return []
             continue
+
+
+def remove_pubchem_smiles(smiles_list: List[str]) -> List:
+    """
+    Function for removing PubChem molecules from a list of smiles.
+
+    Args:
+        smiles_list (List): List of SMILES strings.
+
+    Returns:
+        List:  Filtered list of SMILES, all SMILES pointing to PubChem
+            molecules are removed.
+    """
+
+    if not isinstance(smiles_list, list):
+        raise TypeError(f'Please pass list, not {type(smiles_list)}')
+
+    canonicalizer = Canonicalization(sanitize=False)
+
+    filtered = list(filter(lambda x: not query_pubchem(x), smiles_list))
+    # Canonicalize molecules and filter again (sanity check)
+    filtered_canonical = list(
+        filter(lambda x: not query_pubchem(canonicalizer(x)), filtered)
+    )
+    return filtered_canonical
```

Comment:
```suggestion
from itertools import filterfalse

def remove_pubchem_smiles(smiles_list: Iterable[str]) -> List:
    """
    Function for removing PubChem molecules from an iterable of smiles.

    Args:
        smiles_list (Iterable[str]): many SMILES strings.

    Returns:
        List[str]:  Filtered list of SMILES, all SMILES pointing to PubChem
            molecules are removed.
    """
    canonicalizer = Canonicalization(sanitize=False)

    filtered = filterfalse(query_pubchem(x), smiles_list)
    # Canonicalize molecules and filter again (sanity check)
    filtered_canonical = filterfalse(query_pubchem(canonicalizer(x)), filtered)
    return list(filtered_canonical)
```
Similarly one could return the generator `filtered_canonical: Iterator[str]` directly, so one can stay lazy if needed and aggregate to the desired type directly without going through list. Though the need to wrap might be so commonly that it becomes a nuisance.

Reply:
I think it's reasinable to return the iterator instead of a generator since the lists will be small always since PubChem calls are quite expensive time-wise.